app/src/main/java/com/example/androidthings/assistant/LedBlinkThread.java

```diff
@@ -0,0 +1,65 @@
+package com.example.androidthings.assistant;
+
+import com.google.android.things.pio.Gpio;
+
+import java.io.IOException;
+import java.util.Random;
+
+public class LedBlinkThread extends Thread {
+    private final Gpio mLed;
+    private final Random mRandom;
+    private boolean mBlinking = false;
+    private boolean mClose = false;
+
+    public LedBlinkThread(Gpio led) {
+        mLed = led;
+        mRandom = new Random();
+    }
+
+    public void setBlinking(boolean b) {
```

Comment:
I'm not sure if doing a `setBlinking(boolean)` would be the right structure. Instead, doing something like `blink()` handle the logic in the class would be better.

Reply:
Good point. I originally used the boolean on the way that if you passed false it stopped, but I couldn't find an "end of speech" trigger, so I changed the behaviour but did not rename the method.